lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1062,10 +1062,29 @@ void ADContext::emitNondifferentiabilityError(SILInstruction *inst,
   // that is not associated with any source location, we emit a diagnostic at
   // the instruction source location.
   case DifferentiationInvoker::Kind::AutoDiffFunctionInst:
-  case DifferentiationInvoker::Kind::SILDifferentiableAttribute:
+  case DifferentiationInvoker::Kind::SILDifferentiableAttribute: {
+    auto attr = invoker.getSILDifferentiableAttribute();
     diagnose(opLoc,
              diag.getValueOr(diag::autodiff_expression_is_not_differentiable));
+    if (auto declContext = attr.second->getDeclContext()) {
```

Comment:
The local convention prefers explicit pointer.
```suggestion
    if (auto *declContext = attr.second->getDeclContext()) {
```